indico/modules/events/persons/controllers.py

```diff
@@ -151,17 +148,38 @@ def _checkParams(self, params):
 
     def _process(self):
         person_ids = request.form.getlist('person_id')
-        recipients = {p.email for p in self._find_event_persons(person_ids) if p.email}
-        form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients))
+        recipients = self._find_event_persons(person_ids)
+        if request.args.get('not_invited_only') == '1':
+            recipients = {p for p in recipients if p.invited_dt is None}
```

Comment:
Why not add an argument for this to `_find_event_persons`? You can use `EventPerson.invited_dt.is_(None)` in a SQLAlchemy filter so you don't even query those persons from the DB